### ISSUE TITLE
[Dev] Stop commenting "Closes .." on PR, comment on issue instead

### ISSUE THREAD

**Issue by jorisvandenbossche** [apache/arrow#33863]
### Describe the enhancement requested

It's maybe two separate changes, but:

- The PR workflow both updates the first comment to add a "Closes .." and adds a comment with the same text. I think just updating the top comment is sufficient (the notification from the extra comment isn't useful, and for the rest both updates have the same content).
- On the other hand, what I personally would find useful is that we add a comment to the issue instead that says something like "A PR was opened that would close this issue: ...". 
  If you are subscribed to an issue, you can miss that there is a PR for it until that PR gets merged / the issue closed (on JIRA, you got such a notification)

### Component(s)

Developer Tools

**Comment by assignUser** [comment]
I agree that the extra comment is superflous, I think we currently use it as a guard?

Regarding your second point: once the PR has the `Closes ...` it will show up in the github ui but that will not generate a notification. I personally don't have a preference to but if a notification would be helpful to keep track of stuff I won't opose adding such a comment.

**Comment by rok** [comment]
> I agree that the extra comment is superflous, I think we currently use it as a guard?

I vaguely remember a regex that searched for this in Jira comments to do something. But I think it is no longer used.
